pandas/io/tests/test_common.py

```diff
@@ -139,3 +139,10 @@ def test_next(self):
             self.assertEqual(next_line.strip(), line.strip())
 
         self.assertRaises(StopIteration, next, wrapper)
+
+    def test_cparser_error_depr(self):
+        # see gh-12665
+        with tm.assert_produces_warning(DeprecationWarning,
+                                        check_stacklevel=False):
+            with tm.assertRaises(common.ParserError):
+                raise common.CParserError()
```

Comment:
A common usage will also be:

```
from pandas.io.common import CParserError

try:
    ....
except CParserError:
    ....
```

but I don't think this will currently warn for such use cases?